cpp/src/parquet/arrow/reader.cc

```diff
@@ -270,6 +273,20 @@ class FileReaderImpl : public FileReader {
       records_to_read +=
           reader_->metadata()->RowGroup(row_group)->ColumnChunk(i)->num_values();
     }
+#ifdef ARROW_WITH_OPENTELEMETRY
+    std::string column_name = reader_->metadata()->schema()->Column(i)->name();
+    std::string phys_type = TypeToString(reader_->metadata()->schema()->Column(i)->physical_type());
+    auto span = ::arrow::internal::tracing::GetTracer()->GetCurrentSpan();
+    ::arrow::util::tracing::Span childspan;
+    ::arrow::util::tracing::Span parentspan;
+    parentspan.Set(::arrow::util::tracing::Span::Impl{span});
```

Comment:
This is a confusing way to create a parent span.  Why does `GetCurrentSpan()` not return something we can use?  Why do we need `parentspan`?

Reply:
Looking at this again, it seems superfluous. I'll remove it and just start a span the normal way. That should already take care of setting the current span as the parent automatically.